Uploader/upload/TeamscaleUpload.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using NLog;
+
+/// <summary>
+/// Uploads trace files to Teamscale.
+/// </summary>
+internal class TeamscaleUpload : IUpload
+{
+    private static readonly Logger logger = LogManager.GetCurrentClassLogger();
+
+    private readonly HttpClient client;
+
+    private readonly TeamscaleServer server;
```

Comment:
you can initialize the vars here directly...

Reply:
I only changed the http client since the rest depends on stuff passed to the constructor